client/src/component/Sign-Up-Page.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+
+class SignUpHacker extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            FirstName: '',
+            LastName: '',
+            myEmail: '',
+            phoneNumber: '',
+            University: '',
+            HighSchool: '',
+            Birthday: '',
+            Gender: '',
+            Ethnicity: '',
+            Major: '',
+            CollegeYear: '',
+            ShirtSize: '',
+            myDiet: '',
+            myAllergies: '',
+            resume: '',
+            gitHub: '',
+            LinkedIn: '',
+            Devpost: '',
+        };
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+}
+
+<form>
+    {/* First Name */}
+    <div>
+        <label>First Name</label>
+        <input type="text" name="FirstName" ref="name" placeholder="First Name" />
```

Comment:
```suggestion
        <input type="text" name="FirstName" ref="name" placeholder="First Name" />
```

I believe something is wrong with the `ref`. Like you implemented it incorrectly. Take a look at the [React docs](https://reactjs.org/docs/refs-and-the-dom.html). 

Also you can check that the data is being passed to the super prop using a browser extension called [Reactjs devtools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi).